include/swift/Syntax/SyntaxKinds.def

```diff
@@ -106,6 +109,9 @@ SYNTAX(FunctionParameter, Syntax)
 SYNTAX(DeclModifier, Syntax)
 SYNTAX(DeclModifierList, Syntax)
 
+SYNTAX_COLLECTION(TypeArgumentList, TupleTypeElementSyntax)
```

Comment:
👍 This reveals an issue with the `FunctionTypeSyntax` that means it'll need its own `ElementSyntax` and collection.